src/llmq/quorums.cpp

```diff
@@ -173,6 +178,154 @@ void CQuorum::StartCachePopulatorThread(std::shared_ptr<CQuorum> _this)
     _this->cachePopulatorThread.detach();
 }
 
+size_t CQuorum::GetQuorumRecoveryStartOffset(const CBlockIndex* pIndex) const
+{
+    const size_t nActiveQuorums = params.signingActiveQuorumCount + 1;
+    const std::vector<CQuorumCPtr> vecQuorums = quorumManager->ScanQuorums(qc.llmqType, pIndex, nActiveQuorums);
+    assert(vecQuorums.size() > 0);
+    std::set<uint256> setAllTypeMembers;
+    for (auto& pQuorum : vecQuorums) {
+        auto& vecValid = pQuorum->qc.validMembers;
+        for (size_t i = 0; i< vecValid.size(); ++i) {
+            if (vecValid[i]) {
+                setAllTypeMembers.emplace(pQuorum->members[i]->proTxHash);
+            }
+        }
+    }
+    std::vector<uint256> vecAllTypeMembers{setAllTypeMembers.begin(), setAllTypeMembers.end()};
+    std::sort(vecAllTypeMembers.begin(), vecAllTypeMembers.end());
+    size_t nMyIndex{0};
+    for (size_t i = 0; i< vecAllTypeMembers.size(); ++i) {
+        if (activeMasternodeInfo.proTxHash == vecAllTypeMembers[i]) {
+            nMyIndex = i;
+            break;
+        }
+    }
+    return nMyIndex % qc.validMembers.size();
+}
+
+void CQuorum::StartQuorumDataRecoveryThread(std::shared_ptr<CQuorum> _this, const CBlockIndex* pIndex, uint16_t nDataMaskIn)
+{
+    if (_this->fQuorumDataRecoveryThreadRunning) {
+        LogPrint(BCLog::LLMQ, "CQuorum::%s -- Already running\n", __func__);
+        return;
+    }
+    _this->fQuorumDataRecoveryThreadRunning = true;
+
+    std::thread([_this, pIndex, nDataMaskIn]() {
+        const auto& strThreadName = "dash-q-recovery";
+        RenameThread(strThreadName);
+
+        size_t nTries{0};
+        uint16_t nDataMask{nDataMaskIn};
+        int64_t nTimeLastSuccess{Params().NetworkIDString() == CBaseChainParams::REGTEST ? GetAdjustedTime() : 0};
```

Comment:
Good point 😄 It's a leftover to have a defined start of recovery in tests from an earlier iteration where i had an extra thread running to check each block instead of the "Call `TriggerQuorumDataRecoveryThreads()` in `UpdateBlockTip()` approach. Dropped in 9cf00afa967ef77e026e59030b1c0dd51c044c43

Reply:
Hmm actually.. it might still be required after a reindex to to definitely avoid false positive trigger e.g. here https://github.com/dashpay/dash/pull/3964/files#diff-f0bc67db96f85a2aed8c6e44432c8767d8d6629e7a1972012174cf7d7bba01a7R142, no? 

Seems like dropping it in 9cf00af is also the reason for this failure https://gitlab.com/dashpay/dash/-/jobs/996444551? Which should be fixed by 6558aad though.